core/pta/attestation.c

```diff
@@ -55,21 +55,43 @@ static int cmp_regions(const void *a, const void *b)
 	return memcmp((void *)r1->va, (void *)r2->va, r1->size);
 }
 
-static TEE_Result hash_update(void *ctx, struct vm_region *r, uint32_t mode)
+#define TO_LE64(x) TEE_U64_BSWAP(TEE_U64_TO_BIG_ENDIAN(x))
+
+static TEE_Result hash_update_mode_tags(void *ctx, struct vm_region *r)
 {
-	struct file *f = NULL;
+	uint64_t offs_and_size[2] = { 0, TO_LE64(r->mobj->size) };
+	struct fobj *fobj = r->mobj->ops->get_fobj(r->mobj);
+	struct file *f = to_file_may_fail(r->mobj);
+	TEE_Result res = TEE_SUCCESS;
+	unsigned int poffs = 0;
 	uint8_t *tag = NULL;
 
+	/* Regions we're interested in should all have a struct file */
+	assert(f);
+	file_lock(f);
```

Comment:
This can probably be moved down to only cover the call to `file_find_page_offset()`.

Reply:
OK